src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala

```diff
@@ -211,9 +211,16 @@ object DruidBeamMaker
     val tsUtc = ts.withZone(DateTimeZone.UTC)
 
     val cycleBucket = segmentGranularity match {
-      case Granularity.MINUTE => tsUtc.minuteOfHour.get
-      case Granularity.HOUR => tsUtc.hourOfDay.get
-      case Granularity.DAY => tsUtc.dayOfMonth.get
+      case Granularity.MINUTE => tsUtc.minuteOfHour().get
+      case Granularity.FIVE_MINUTE => tsUtc.minuteOfHour().get
+      case Granularity.TEN_MINUTE => tsUtc.minuteOfHour().get
+      case Granularity.FIFTEEN_MINUTE => tsUtc.minuteOfHour().get
+      case Granularity.HOUR => tsUtc.hourOfDay().get
+      case Granularity.SIX_HOUR => tsUtc.hourOfDay().get
+      case Granularity.DAY => tsUtc.dayOfMonth().get
+      case Granularity.WEEK => tsUtc.weekOfWeekyear().get
+      case Granularity.MONTH => tsUtc.monthOfYear().get
+      case Granularity.YEAR => if (tsUtc.yearOfCentury().get == 100) 0 else tsUtc.yearOfCentury().get
```

Comment:
Is the special case needed. As far as I know, if we are using the default ISOChronology, yearOfCentury should always be between 0 and 99